### DISCUSSION TITLE
reorganize settings menu

### DISCUSSION THREAD

**Discussion by pgScorpio** [jamulussoftware/jamulus#2507] [UI Related]
I experience that a lot of new jamulus users have trouble finding the needed settings. Especially those in the "Advanced Settings" menu since they think a newbie should not mess with "Advanced settings", though they might really need them.
And even, as an experienced user, I find myself switching between tabs a lot during an initial setup.

My new proposal is to get rid of the (not so) "Advanced" features tab and split up in "User", "Audio" and "Network" tabs and just rearrange the options like this:

![UserSettings](https://user-images.githubusercontent.com/9221087/158077491-f16af417-9bf7-4a3c-94f7-06f649c514ca.png)
![AudioSettings](https://user-images.githubusercontent.com/9221087/158077496-06f54dac-8815-46fb-9929-944ffc7929f7.png)
![NetworkSettings](https://user-images.githubusercontent.com/9221087/158077500-7be4ec35-7508-49e7-a9e6-38116a99c6d2.png)

Also we use a more "landscape" layout now, which fits better on small screen devices like phone and tablet as used with the android version.

edit: BTW The value indicators for Balance are misleading too. Showing a "R" value when panning to the left and showing a "L" when panning to the right.

Please comment with your remarks and ideas....

**Comment by jujudusud** [comment]
Nice Works !
I think it's interesting to have "Audio Mode", "Audio Stream Rate" and all the "Network setup" stuff at the same place.
The tabs, in this case, should be rename "interface setup" instead of "Audio setup".

**Reply by pgScorpio** [reply]
> The tabs, in this case, should be rename "interface setup" instead of "Audio setup".

That would be "Audio Device" to be consistant, but it's not only the audio device on this tab, but all audio related settings...

**Reply by gilgongo** [reply]
"Sound setup"? Yes, other settings like buffers affect sound, but from a top level, "sound" is universally recognised as a thing separate from "user" and "network".

**Comment by ann0see** [comment]
Tagging @dcorson-ticino-com 
@mulyaj here.


@pgScorpio first if all: thanks for another UI design proposal. Initially we decided against splitting audio and network since they are largely coupled (personally if sound is bad, I‘d change audio buffer delay, network buffers, ASIO Driver settings and don’t want to jump between two tabs). I think there are some settings a user fiddles around a lot:
1. Buffer delay
2. ASIO settings (opening ASIO4ALL)
3. Device changing setup
4. Maybe: mono/stereo settings
They should be quickly reachable

I think the current behaviour can and should be improved, but I‘m not quite sure how to re-arrange it. 


Which settings aren’t advanced in your opinion (probably that largely depends on the use case)?

**Reply by pgScorpio** [reply]
Concerning switching tabs:
User settings are normally a one time setup.
Device selection and device settings are normally never changed once ok, so that leaves Network settings for tweeking current connection problems.

> Which settings aren’t advanced in your opinion (probably that largely depends on the use case)?

All depends on what you call "Advanced". Normally "Advanced" means tweaking for experienced users only.
So in my opinion **none** of the "Advanced" settings belong to this category.

"New Client Level" and "Custom Directories" are even very basic. And why is "Balance" an advanced feature?

In the current situation I find myself switching tabs a lot more. Especially now when I am debugging sound ;=) so that where this idea originated.

In my opinion the settings should be logically aranged, so any user (and especially new users) can intuitively find the wanted setting.

**Reply by ann0see** [reply]
> Device selection and device settings are normally never changed once ok

my feeling is still that this should be always visible since I think it’s a core feature of an audio application to see which device is being used. 

> And why is "Balance" an advanced feature?

* By assumption audio interfaces let you control the input gain for left/right signal
* non advanced users (ASIO4ALL) are probably only using one input
* Software level wise there are usually ways to control gain

I don’t know many people using this feature…

> "New Client Level" and "Custom Directories" are even very basic. 

Partly agree. New client level is a set and forget feature; same for custom directories. 

I think the feeling was that in order to get started (=hear sound) with Jamulus nobody actually needs custom directories. Once they got set up and use the app for private rehearsals or WorkdJam presumably they’re directed to adding the custom directory.

**Comment by dcorson-ticino-com** [comment]
This is more difficult than it looks.
On one hand newbies will always have problems, no matter how the settings are distributed, on the other many settings you will only touch once, while there are others that may (have) to be used more often over time.  As ann0see said this was considered in making the separation we have now so that settings that may be necessary to be used more often because of the vagarities of the Internet and will be used together to find the sweet spot for audio quality are placed together.
I don't really see that things are logically better, custom directories really has nothing to do with my user settings and audio stream rate is a network information.  
Maybe we should change the name "Advanced" to "Other" if people are really scared away because of that.
BTW Although I use Jamulus several times a week I don't think that I have ever opened the "profile" or "advanced" tab in the last 6 months, but changing buffer and quality settings is sometimes necessary, all together in one little tab.
Oh, and the high and thin window style was done at the request of potential Android users if I remember correctly, so I don't know what to think there.

**Reply by pgScorpio** [reply]
> custom directories really has nothing to do with my user settings and audio stream rate is a network information.

In my opinion custom directories is nothing but a user setting.

Audio stream rate may be seen as a network issue, but strongly related to audio mode and quality, which are definitely audio settings in my opinion.

Sure this isn't easy matter, since everybody's use case will be different. Don't forget that a musician and a choir member both look different at this issue, and it also depends on if you just use a fixed server for fixed rehearsels or you switch a lot while attending different events. So ordering by functionality seemed the most logical and intuitive to me.

**Reply by gilgongo** [reply]
I agree. It all very much depends on the user's overall "mental modal" and there will be no one-fits-all. If enough people think that "audio" is also about "network" settings, then we could combine those tabs (making the sliders appear if you un-check "auto" perhaps?) 

BTW we could organise a card sort (I think @mulyaj has done this in the past) to see what the "best fit" organisation is.

**Comment by pgScorpio** [comment]
OK How about if we just dump the "Advanced page" And keep a "Audio/Network" page ? 
Still fits quite nicely.
![UserSettings-2](https://user-images.githubusercontent.com/9221087/159197467-37d0f486-31f8-4304-bf8b-1b11bd304c05.png)

![Audio-NetworkSettings](https://user-images.githubusercontent.com/9221087/159197281-00797cb7-f158-4992-9703-37f698da73af.png)

**Reply by gilgongo** [reply]
Yes, I think that would solve the tab naming problem. Might want to re-arrange it a bit perhaps? Something like this (but with the widths of those panels adjusted in the second screenshot - this is just a cut/paste job)? Various ways of arranging the items though...

![image](https://user-images.githubusercontent.com/4561747/159285805-f14176f2-0c27-4d0c-9c90-b5fd569ea3b3.png)

**Reply by pgScorpio** [reply]
Yes, this arrangement looks quite nice too !
Though I would place Audio Stream Rate in (or under) the Audio Mode panel than. 
Also Feedback Protection might be put under Input Boost, 
Just to keep Audio and Network better separated ?

**Reply by gilgongo** [reply]
Yes, good suggestions - so helping to group it better. More like this maybe then?

![image](https://user-images.githubusercontent.com/4561747/159295224-ad6580ff-caf2-4590-8311-7f4fc500b2e2.png)

**Reply by pgScorpio** [reply]
Yes, that's exactly what I meant!
Maybe it would be nicer to place Audio Stream Rate in the Audio mode group box, but that's trivial I guess.

**Comment by pgScorpio** [comment]
As implemented now in my [repo](https://github.com/pgScorpio/jamulus/tree/autobuild/soundRedesignDev):
![image](https://user-images.githubusercontent.com/9221087/159555626-a8f66ef0-1aa8-48ef-a77e-8ee804532f5f.png)
![image](https://user-images.githubusercontent.com/9221087/159555787-cdf660c3-755a-4a1d-b10a-ff64192d1831.png)

**Reply by gilgongo** [reply]
That looks pretty good to me!
